File: app/src/main/java/com/erikzuo/portfolioandroidapp/data/DataManager.java
package com.erikzuo.portfolioandroidapp.data;

import com.erikzuo.portfolioandroidapp.data.model.Education;
import com.erikzuo.portfolioandroidapp.data.model.MyInfo;
import com.erikzuo.portfolioandroidapp.data.model.Repo;
import com.erikzuo.portfolioandroidapp.data.model.Work;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by YifanZuo on 3/2/18.
 */

public interface DataManager {

    Observable<MyInfo> getMyInfo();

    Observable<List<Repo>> getRepoListApiCall();

    Observable<List<Education>> getEducationList();

    Observable<List<Work>> getWorkList();
}

File: app/src/main/java/com/erikzuo/portfolioandroidapp/data/DataManager.kt
package com.erikzuo.portfolioandroidapp.data

import com.erikzuo.portfolioandroidapp.data.model.*

import io.reactivex.Observable

/**
 * Created by YifanZuo on 3/2/18.
 */

interface DataManager {

    fun getMyInfo(): Observable<MyInfo>

    fun getRepoListApiCall(): Observable<List<Repo>>

    fun getEducationList(): Observable<List<Education>>

    fun getSkillList(): Observable<List<Skill>>
}
